src/Kiota.Builder/Writers/Dart/CodeMethodWriter.cs

```diff
@@ -0,0 +1,885 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Kiota.Builder.CodeDOM;
+using Kiota.Builder.Extensions;
+using Kiota.Builder.OrderComparers;
+using Kiota.Builder.Refiners;
+using static Kiota.Builder.CodeDOM.CodeTypeBase;
+
+namespace Kiota.Builder.Writers.Dart;
+public class CodeMethodWriter : BaseElementWriter<CodeMethod, DartConventionService>
+{
+    public CodeMethodWriter(DartConventionService conventionService) : base(conventionService)
+    {
+
+    }
+    public override void WriteCodeElement(CodeMethod codeElement, LanguageWriter writer)
+    {
+        ArgumentNullException.ThrowIfNull(codeElement);
+        if (codeElement.ReturnType == null) throw new InvalidOperationException($"{nameof(codeElement.ReturnType)} should not be null");
+        ArgumentNullException.ThrowIfNull(writer);
+        if (codeElement.Parent is not CodeClass parentClass) throw new InvalidOperationException("the parent of a method should be a class");
+
+        var returnType = conventions.GetTypeString(codeElement.ReturnType, codeElement);
+        var inherits = parentClass.StartBlock.Inherits != null && !parentClass.IsErrorDefinition;
+        var isVoid = conventions.VoidTypeName.Equals(returnType, StringComparison.OrdinalIgnoreCase);
+        WriteMethodDocumentation(codeElement, writer);
+        WriteMethodPrototype(codeElement, parentClass, writer, returnType, inherits, isVoid);
+        writer.IncreaseIndent();
+
+        HandleMethodKind(codeElement, writer, inherits, parentClass, isVoid);
+        var isConstructor = codeElement.IsOfKind(CodeMethodKind.Constructor, CodeMethodKind.ClientConstructor, CodeMethodKind.RawUrlConstructor);
+
+        if (HasEmptyConstructorBody(codeElement, parentClass, isConstructor))
+        {
+            writer.DecreaseIndent();
+        }
+        else
+        {
+            if (isConstructor && !inherits && parentClass.Properties.Where(x => x.IsOfKind(CodePropertyKind.AdditionalData)).Any() && !parentClass.IsErrorDefinition && !parentClass.Properties.Where(x => x.IsOfKind(CodePropertyKind.BackingStore)).Any())
```

Comment:
```suggestion
            if (isConstructor && !inherits && parentClass.Properties.Where(static x => x.Kind is CodePropertyKind.AdditionalData).Any() && !parentClass.IsErrorDefinition && !parentClass.Properties.Where(static x => x.Kind is CodePropertyKind.BackingStore).Any())
```

ask: static is missing a couple of places for lambdas
nit: using pattern matching instead of calling a method should also be faster